kubetest/e2e.go

```diff
@@ -393,6 +436,8 @@ type deployer interface {
 	IsUp() error
 	SetupKubecfg() error
 	Down() error
+	FederationUp() error
```

Comment:
I started out that way. The reason why I switched to this approach is because federation could be potentially tested using all these deployment methods. I see federation as an optional addon on top of existing deployment methods, of which we only have an implementation for bash scripts.